### DISCUSSION TITLE
Migrating to eslint.config.js from .eslintrc

### DISCUSSION THREAD

**Discussion by staraphd** [eslint/eslint#18919] [Q&A]
This is in continuation to https://github.com/eslint/eslint/discussions/18915 which was closed and my org doesn't allow adding new packages.

I followed the migration guide https://eslint.org/docs/latest/use/configure/migration-guide and created the new config file using 
` npx @eslint/migrate-config .eslintrc` 
The migration guide says that new config file is named `eslint.config.js`, but the migrate-config package created a file named `eslint.config.mjs`. Should I be renaming it to `eslint.config.js`?

My existing `.eslintrc` looked like so
```
{
    "root": true,
    "parser": "@typescript-eslint/parser",
    "plugins": [
      "@typescript-eslint"
    ],
    "extends": [
      "eslint:recommended",
      "plugin:@typescript-eslint/eslint-recommended",
      "plugin:@typescript-eslint/recommended"
    ],
    "rules": { }
}
```
And `.eslintignore` looked like so
```
node_modules
dist
test
fix
```
And `package.json` looked like so
```
  "scripts": {
    "lint": "eslint . --ext .ts",
```
Is the modified `eslint.config.mjs` below correct?

```
export default [{
    files: ["src/**/*.ts"],
    ignores: ["**/node_modules", "**/dist", "**/test", "**/fix"],
}, ...compat.extends(
    "eslint:recommended",
    "plugin:@typescript-eslint/eslint-recommended",
    "plugin:@typescript-eslint/recommended",
), {
    plugins: {
        "@typescript-eslint": typescriptEslint,
    },

    languageOptions: {
        parser: tsParser,
    },   
rules: {},
}];
```
When running `npm run lint`, it is not ignoring the directories specified in `ignores`
I added the `files` key as per the existing script `"lint": "eslint . --ext .ts"` in `package.json`. Is the `files` key redundant and should I remove it for `ignores` to work?

Is the modified `package.json` correct?

```
  "scripts": {
    "lint": "eslint",
```
Should it rather be the following?

```
  "scripts": {
    "lint": "eslint .",
```

**Comment by nzakas** [comment]
> The migration guide says that new config file is named `eslint.config.js`, but the migrate-config package created a file named `eslint.config.mjs`. Should I be renaming it to `eslint.config.js`?

No, all of `js` extensions work.

> Is the `files` key redundant and should I remove it for `ignores` to work?

`ignores` must be in an object [by itself](https://eslint.org/docs/latest/use/configure/configuration-files#globally-ignoring-files-with-ignores) without `files`. Did you modify the file created by the migrator? The migrator should create the correct format for you, including converting your `.eslintignore` file into a correct `ignores` object.
